tests/unit/ContentBlocksResolverTest.php

```diff
@@ -136,4 +136,108 @@ static function ( $parsed_block ) {
 		$this->assertEquals( 4, count( $parsed_blocks ) );
 		$this->assertEquals( $allowed, $actual_block_names );
 	}
+
+	/**
+	 * Test the wpgraphql_content_blocks_pre_resolve_blocks filter.
+	 */
+	public function test_pre_resolved_blocks_filter_returns_non_null() {
+		add_filter(
+			'wpgraphql_content_blocks_pre_resolve_blocks',
+			static function ( $blocks, $node, $args, $allowed_block_names ) {
+				return [
+					[
+						'blockName' => 'core/paragraph',
+						'attrs'     => [ 'content' => 'Test content' ],
+					],
+				];
+			},
+			10,
+			4
+		);
+
+		$post_id = self::factory()->post->create( [ 'post_content' => '' ] );
+		$post    = new Post( get_post( $post_id ) );
+
+		$resolved_blocks = $this->instance->resolve_content_blocks( $post, [] );
+		// The filter should return a block.
+		$this->assertCount( 1, $resolved_blocks );
+		$this->assertEquals( 'core/paragraph', $resolved_blocks[0]['blockName'] );
+		$this->assertEquals( 'Test content', $resolved_blocks[0]['attrs']['content'] );
+	}
+
+	/**
+	 * Tests content retrieval from a post node.
+	 */
+	public function test_content_retrieved_from_post_node() {
+		$post_id         = self::factory()->post->create(
+			[
+				'post_content' => '<!-- wp:paragraph --><p>Content</p><!-- /wp:paragraph -->',
+			]
+		);
+		$post            = new Post( get_post( $post_id ) );
+		$resolved_blocks = $this->instance->resolve_content_blocks( $post, [] );
+
+		// The block should be resolved from the post node.
+		$this->assertCount( 1, $resolved_blocks );
+		$this->assertEquals( 'core/paragraph', $resolved_blocks[0]['blockName'] );
+	}
+
+	/**
+	 * Tests that an empty array is returned when the post content is empty.
+	 */
+	public function test_returns_empty_array_for_empty_content() {
+		$post_id = self::factory()->post->create( [ 'post_content' => '' ] );
+		$post    = new Post( get_post( $post_id ) );
+
+		$resolved_blocks = $this->instance->resolve_content_blocks( $post, [] );
+
+		// The block should be resolved from the post node.
+		$this->assertIsArray( $resolved_blocks );
+		$this->assertEmpty( $resolved_blocks );
+	}
+
+	/**
+	 * Tests that the wpgraphql_content_blocks_allowed_blocks filter is applied.
+	 */
```

Comment:
What's different about this vs the preexisting `test_resolve_content_blocks_filters_blocks_not_from_allow_list()`?